sky/packages/sky/lib/src/rendering/editable_paragraph.dart

```diff
@@ -0,0 +1,115 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:sky/painting.dart';
+import 'package:sky/src/rendering/box.dart';
+import 'package:sky/src/rendering/object.dart';
+import 'package:sky/src/rendering/paragraph.dart';
+import 'package:sky/src/rendering/proxy_box.dart' show SizeChangedCallback;
+
+const _kCursorGap = 1.0;
+const _kCursorHeightOffset = 2.0;
+const _kCursorWidth = 1.0;
+
+/// A render object used by EditableText widgets.  This is similar to
+/// RenderParagraph but also renders a cursor and provides support for
+/// scrolling.
+class RenderEditableParagraph extends RenderParagraph {
+
+  RenderEditableParagraph({
+    TextSpan text,
+    Color cursorColor,
+    bool showCursor,
+    this.onContentSizeChanged,
+    Offset scrollOffset
+  }) : _cursorColor = cursorColor,
+       _showCursor = showCursor,
+       _scrollOffset = scrollOffset,
+       super(text, allowLineWrap: false);
+
+  Color _cursorColor;
+  bool _showCursor;
+  SizeChangedCallback onContentSizeChanged;
+  Offset _scrollOffset;
+
+  Size _contentSize;
+
+  Color get cursorColor => _cursorColor;
+  void set cursorColor(Color value) {
+    if (_cursorColor == value)
+      return;
+    _cursorColor = cursorColor;
+    markNeedsPaint();
+  }
+
+  bool get showCursor => _showCursor;
+  void set showCursor(bool value) {
+    if (_showCursor == value)
+      return;
+    _showCursor = showCursor;
+    markNeedsPaint();
+  }
+
+  Offset get scrollOffset => _scrollOffset;
+  void set scrollOffset(Offset scrollOffset) {
+    _scrollOffset = scrollOffset;
+    markNeedsPaint();
+  }
+
+  double getMinIntrinsicWidth(BoxConstraints constraints) {
+    return constraints.constrainWidth(
+      super.getMinIntrinsicWidth(constraints) + _kCursorGap + _kCursorWidth
```

Comment:
This isn't accurate. For example, consider the case where the text looks like this:

```
Hello Nice Big
World|
```

...where the cursor is the vertical bar. The minimum intrinsic width in this case is unaffected by the cursor.

If you only support single-line text, such that there's only one intrinsic width, then just have a single function that you call from both getMinIntrinsicWidth and getMaxIntrinsicWidth (and assert that textPainter.minContentWidth and textPainter.maxContentWidth are the same, and document that this is only valid because we don't wrap).